overlay/sbin/k3s-upgrade

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+if [ $(whoami) != "root" ]
+then
+    echo "This script must be run as root."
+    exit 1
+fi
+
+PROC=$(uname -m)
+
+# Take a parameter of the version number (i.e. v0.4.0) if it is given, otherwise use latest
+if [ -z $K3S_VERSION ]
+then
+    K3S_VERSION=$(curl -sL api.github.com/repos/rancher/k3s/releases/latest | jq .tag_name -r)
+fi
+
+if [ $PROC == "x86_64" ]
+then
+    ARCH="amd64"
+elif [ $PROC == "aarch64" ]
+then
+    ARCH="arm64"
+elif [ $PROC == "armv7l" ]
+then
+    ARCH="armhf"
+else
+    echo "Unsupported CPU architecture."
+    exit 1
+fi
+
+K3S_URL="https://github.com/rancher/k3s/releases/download/${K3S_VERSION}/k3s-${ARCH}"
```

Comment:
On amd64, the release binary is called 'k3s' with no arch suffix, so we need a special case to handle this.

`if [ $ARCH == "amd64" ]
then
     K3S_URL="https://github.com/rancher/k3s/releases/download/${K3S_VERSION}/k3s"
else
     K3S_URL="https://github.com/rancher/k3s/releases/download/${K3S_VERSION}/k3s-${ARCH}"
fi`